tests/SchemeSpec.hs

```diff
@@ -159,22 +169,56 @@ testUserSig = do
         wrongScheme = ED25519
     [kp] <- mkKeyPairs [apiKP]
     sig <- sign kp hsh
-    let myUserSig = UserSig wrongScheme (toB16Text pubBS) addr (toB16Text sig)
-    (verifyUserSig hsh myUserSig) `shouldBe` False
+    let myUserSig = UserSig (toB16Text sig)
+        mySigner = Signer wrongScheme (toB16Text pubBS) addr
+    (verifyUserSig hsh myUserSig mySigner) `shouldBe` False
 
 
 
 
   it "provides default ppkscheme when one not provided" $ do
     let sigJSON = A.object ["addr" .= String "SomeAddr", "pubKey" .= String "SomePubKey",
                             "sig" .= String "SomeSig"]
-        sig = UserSig defPPKScheme "SomePubKey" "SomeAddr" "SomeSig"
+        sig = UserSig "SomeSig"
     (fromJSON' sigJSON) `shouldBe` (Right sig)
 
 
 
 
   it "makes address field the full public key when one not provided" $ do
     let sigJSON = A.object ["pubKey" .= String "SomePubKey", "sig" .= String "SomeSig"]
-        sig = UserSig defPPKScheme "SomePubKey" "SomePubKey" "SomeSig"
+        sig = UserSig "SomeSig"
     (fromJSON' sigJSON) `shouldBe` (Right sig)
+
+
+testSigNonMalleability :: Spec
+testSigNonMalleability = do
+  it "fails when invalid signature provided for signer specified in the payload" $ do
+    let (pub1, priv1, addr1, scheme1) = someETHPair
+        apiKP1 = ApiKeyPair priv1 (Just pub1) (Just addr1) (Just scheme1)
+        (pub2, priv2, addr2, scheme2) = someED25519Pair
+        apiKP2 = ApiKeyPair priv2 (Just pub2) (Just addr2) (Just scheme2)
+    kp1 <- mkKeyPairs [apiKP1]
+    wrongPair <- mkKeyPairs [apiKP2]
+    
+    cmdWithWrogSig <- mkCommand' wrongPair $ toExecPayload kp1 "(somePactFunction)"
+    (verifyCommand cmdWithWrogSig :: ProcessedCommand () ParsedCode)
+      `shouldSatisfy`
+      (\t -> case t of
```

Comment:
make a utility function out of this (code de-dupe with line 222)

Reply:
Refactored a bit in the recent commit.